feat(vim): create vim extension (#3118)
felangel/bloc: extensions/vim/README.md

# vim-bloc-plugin

> This plugin was designed to be similar the [vscode extension](https://github.com/felangel/bloc/tree/master/extensions/vscode).

Reviewer: ```suggestion
This plugin was designed to be similar as the [vscode extension](https://github.com/felangel/bloc/tree/master/extensions/vscode).
```

Generating the Flutter Bloc or Cubit files.

## Installation

> Assuming you are using vim-plug as plugin manager, you can just put it in yout `init.vim`

Reviewer: ```suggestion
Assuming you are using vim-plug as plugin manager, you can just put it in your `init.vim`
```

```bash
Plug 'felangel/bloc', { 'rtp': 'extension/vim' }
```

run `:PlugInstall`, after that, you must source your `.vimrc`

## Usage

> After your had installed the plugin, it enable two new commands

Reviewer: ```suggestion
After your have installed the plugin, it enables two new commands
```

```bash
:Bloc <BlocName> <DestinationFolder>
```

> - BlocName: Is the name will be used to generate all bloc files and classes

Reviewer: ```suggestion
- BlocName: Is the name that will be used to generate all bloc files and classes
```

    - **This name must be a CamelCase**
- DestinationFolder: Is the path to generate the files, like: `lib/posts`
    - **Don't start with `/`**

```bash
:Cubit <CubitName> <DestinationFolder>
```

> - CubitName: Is the name will be used to generate all cubit files and classes

Reviewer: ```suggestion
- CubitName: Is the name that will be used to generate all cubit files and classes
```
